### ISSUE TITLE
Should we enable strict-null-checks in our code ? 

### ISSUE THREAD

**Issue by shsuman** [microsoft/vscode-tools-for-ai#682]
What is strict null checking ? 
From typescriptlang.org:
In strict null checking mode, the null and undefined values are not in the domain of every type and are only assignable to themselves and any (the one exception being that undefined is also assignable to void). So, whereas T and T | undefined are considered synonymous in regular type checking mode (because undefined is considered a subtype of any T), they are different types in strict type checking mode, and only T | undefined permits undefined values. The same is true for the relationship of T to T | null

What we lose ?
Basically, once we enable this, we won't be able to abuse 'undefined' as we do right now. We cannot just return from a function abruptly which has a non null return type.

What we gain ?
Should help us greatly in catching null values that usually escapes compiler and are thrown at runtime. Our scenario fits quite right here as VS Code uses undefined extensively to denote cancellation of an operation and so do we.

Interested to know what @greazer thinks about this :P

**Comment by sevillal** [comment]
Dup of https://github.com/microsoft/vscode-tools-for-ai/issues/113
